Added a new pattern folder and its related documents “Custom Filter”

sdash-redhat opened the pull request:
The current pull request contains a new pattern design submission "Custom Filter".

catrobson commented on pattern-library/forms-and-controls/custom-filter/design/overview.md:
```diff
@@ -0,0 +1,21 @@
+## Use Cases
+
+This pattern can be used where the PatternFly Filter is used to filter a data set on the view. The user while applies different filter criteria in a view, the same can be saved with a name for later usage.
+
```
I think we want this to really hone in on the "saving" part of the actions, so a suggested reword.  What do you think?

"This pattern should be used where users want to save filters they have built out for future use."

sdash-redhat replied:
Makes sense. It goes well with the name and indicates the purpose of the filter clearly!

catrobson commented on pattern-library/forms-and-controls/custom-filter/design/overview.md:
```diff
@@ -0,0 +1,21 @@
+
+
+### Saving a Custom Filter
+![Saving a Filter](img/customfilter-page2.png)
```
We discovered during implementation of this into patternfly-ng that putting the save action on the right hand side conflicts with some other patterns where we should number of pages and number of results on this area.  It was suggested that we should put the Save on the left after the "Clear Filter" action.  Can you try an update to the design with this?  You can see the problem as discussed by Dan here:  https://github.com/patternfly/patternfly-ng/pull/155

sdash-redhat replied:
Reviewed the thread patternfly/patternfly-ng#155. It's clear now. 
Will make the changes.

sdash-redhat commented on pattern-library/forms-and-controls/custom-filter/design/overview.md:
```diff
@@ -0,0 +1,21 @@
+This template should provide high level information on the design, to inform the user, at a glance, if this pattern is a fit for their needs.
+
+# Custom Filter
```
Agree. Will change the name and the folder attributes

sdash-redhat commented on pattern-library/forms-and-controls/custom-filter/design/overview.md:
```diff
@@ -0,0 +1,21 @@
+
+## Overview  
+
+Custom Filter helps the user in saving any applied filter criteria with a name so that at a later point of time he can access the same from a list of saved such filter criteria and apply for desired the result set.
```
This is helpful :)

sdash-redhat commented:
Updated the files. Renamed the "Custom filter" folder to "Saved Filters" . Updated content, file names and images accordingly along with the changes prescribed. 
Please review. 
Best
Samir

dlabrecq commented:
FYI, the Travis build failed with the following message.

"Error: This site.md file references a invalid design.md file:
 site.md: pattern-library/forms-and-controls/saved-filter/site.md
 design: pattern-library/{category-name}/{pattern-name}/design/design.md"

sdash-redhat commented:
Added the section describing the 'delete' interaction as discussed.

mcarrano commented:
This PR has been inactive for over 6 months and I will close. If this work picks up again in the future, I recommend opening a new PR.
